src/api/app/services/workflows/yaml_to_workflows_service.rb

```diff
@@ -31,8 +35,11 @@ def parse_workflows_file(file_path)
       target_repository_full_name = @scm_webhook.payload.values_at(:target_repository_full_name, :path_with_namespace).compact.first
       scm_organization_name, scm_repository_name = target_repository_full_name.split('/')
 
+      workflows_file_content = File.read(file_path)
+      track_placeholder_variables(workflows_file_content)
+
       # Mapping the placeholder variables to their values from the webhook event payload
-      format(File.read(file_path),
+      format(workflows_file_content),
              SCM_ORGANIZATION_NAME: scm_organization_name,
```

Comment:
It would be nice if we could rely on `SUPPORTED_PLACEHOLDER_VARIABLES` here to do the mapping. Someone has an idea how? Otherwise we need to keep this `format(...)` call and `SUPPORTED_PLACEHOLDER_VARIABLES` in sync.

Reply:
What about this?
```
irb(main):001:0> SUPPORTED_PLACEHOLDER_VARIABLES = ['SCM_ORGANIZATION_NAME', 'SCM_REPOSITORY_NAME', 'SCM_PR_NUMBER', 'SCM_COMMIT_SHA'].freeze
=> ["SCM_ORGANIZATION_NAME", "SCM_REPOSITORY_NAME", "SCM_PR_NUMBER", "SCM_COMMIT_SHA"]
irb(main):002:0> values = ['organization_name', 'repository_name']
=> ["organization_name", "repository_name"]    
irb(main):003:0> SUPPORTED_PLACEHOLDER_VARIABLES.zip(values).to_h
=> {"SCM_ORGANIZATION_NAME"=>"organization_name", "SCM_REPOSITORY_NAME"=>"repository_name", "SCM_PR_NUMBER"=>nil, "SCM_COMMIT_SHA"=>nil}
irb(main):004:0>
```

Reply:
Damn, it's merged already... We'll fix it here https://trello.com/c/fj2rk8NT/205-make-it-easier-to-maintain-the-placeholder-variables